pkg/compose/build.go

```diff
@@ -189,11 +189,14 @@ func (s *composeService) getLocalImagesDigests(ctx context.Context, project *typ
 		images[name] = info.ID
 	}
 
-	for _, s := range project.Services {
-		imgName := getImageName(s, project.Name)
+	for i := range project.Services {
+		imgName := getImageName(project.Services[i], project.Name)
 		digest, ok := images[imgName]
 		if ok {
-			s.CustomLabels[api.ImageDigestLabel] = digest
+			if project.Services[i].CustomLabels == nil {
+				project.Services[i].CustomLabels = make(map[string]string)
+			}
+			project.Services[i].CustomLabels[api.ImageDigestLabel] = digest
```

Comment:
Labels property propose an `add` method we should use to add the label

Reply:
This assumes that the added labels didn't exist before, is that accurate?